modules/dhp_connected_devices/app/services/fitbit_service.rb

```diff
@@ -4,14 +4,14 @@
 
 class FitbitService
   def get_connection_status(data)
-    @fitbit_client = data[:fitbit_api]
-    data[:callback_params][:code] && exchange_code_for_token(data[:callback_params][:code]) ? 'success' : 'error'
+    fitbit_client = data[:fitbit_api]
+    data[:callback_params][:code] && exchange_code_for_token({ code: data[:callback_params][:code], client: fitbit_client}) ? 'success' : 'error'
   end
 
   private
 
-  def exchange_code_for_token(code)
-    response = @fitbit_client.get_token(code)
+  def exchange_code_for_token(data)
+    response = data[:client].get_token(data[:code])
     response.status == 200
   end
```

Comment:
@yonidoronpeters what do you think of this change?

Reply:
Line 8 is pretty complex, and there's inconsistency between the fitbit client naming which is a bit confusing. It will work, but not super clean. 
Ideally, we would move the fitbit client initialization from the controller into this service. And then delegate to the service when we need to call fitbit with the client. But that will result in some changes to tests. What do you think?

Reply:
@yonidoronpeters good point. We'll do that.